components/br-stream/src/codec.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+use bytes::{Buf, Bytes};
+use std::io::prelude::*;
+use std::io::Cursor;
+
+pub struct Encoder;
+
+impl Encoder {
+    // TODO use a more efficent encode kv event format
+    pub fn encode_event(key: &[u8], value: &[u8]) -> Vec<u8> {
+        let mut buf = vec![];
```

Comment:
Maybe prealloc this.
```suggestion
        let mut buf = Vec::with_capacity(key.len() + value.len() + 8);
```